File: gameSrc/commands/activate_power.js
cwt.Action.mapAction({
  key: "activatePower",

  condition: function () {
    return cwt.CO.canActivatePower(cwt.Gameround.turnOwner, cwt.CO.POWER_LEVEL_COP);
  },

  hasSubMenu: true,
  prepareMenu: function (data) {

    data.menu.addEntry("cop");
    if (cwt.CO.canActivatePower(cwt.Gameround.turnOwner, cwt.CO.POWER_LEVEL_SCOP)) {
      data.menu.addEntry("scop");
    }
  },

  toDataBlock: function (data, dataBlock) {
    dataBlock.p1 = (data.action.selectedSubEntry === "cop" ? cwt.CO.POWER_LEVEL_COP : -1);
    dataBlock.p1 = (data.action.selectedSubEntry === "scop" ? cwt.CO.POWER_LEVEL_SCOP : -1);
  },

  parseDataBlock: function (dataBlock) {
    cwt.CO.activatePower(cwt.Gameround.turnOwner,dataBlock.p1);
  }
});

File: gameSrc/org/wolftec/cwt/actions/gameactions/ActivatePower.java
package org.wolftec.cwt.actions.gameactions;

import org.wolftec.cwt.actions.Action;
import org.wolftec.cwt.actions.ActionData;
import org.wolftec.cwt.actions.ActionType;
import org.wolftec.cwt.actions.UserInteractionData;
import org.wolftec.cwt.logic.CommanderLogic;
import org.wolftec.cwt.model.ModelManager;

public class ActivatePower implements Action {

  private ModelManager   model;
  private CommanderLogic co;

  @Override
  public String key() {
    return "activatePower";
  }

  @Override
  public ActionType type() {
    return ActionType.MAP_ACTION;
  }

  @Override
  public boolean condition(UserInteractionData data) {
    return co.canActivatePower(data.actor, CommanderLogic.POWER_LEVEL_COP);
  }

  @Override
  public boolean hasSubMenu() {
    return true;
  }

  @Override
  public void prepareMenu(UserInteractionData data) {
    // TODO
    data.addInfo("cop", false);
    if (co.canActivatePower(data.actor, CommanderLogic.POWER_LEVEL_SCOP)) {
      data.addInfo("scop", false);
    }
  }

  @Override
  public void fillData(UserInteractionData interactionData, ActionData actionData) {
    actionData.p1 = interactionData.actor.id;
    actionData.p2 = interactionData.actionDataCode;
  }

  @Override
  public void invoke(ActionData data) {
    co.activatePower(model.getPlayer(data.p1), data.p2);
  }

}
